### PR TITLE
WIP Feature/implement contract

### PR DESCRIPTION
**Pull request by kimlisa**

### CODE REVIEW

**Comment by kimlisa** [comment]
am i suppose to put the WIP tag on the pull request header? or for each commit? i've been forgetting to put the WIP tag

File: logtag/logtag.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package logtag
```
Comment (faraonc) [inline logtag/logtag.go]: Multiple services will be logging. Please move any logging feature to a new repo. https://github.com/faraonc/hwsc-logging

File: service/service.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	return nil, nil
+}
+
+func (s *Service) ShareDocument (ctx context.Context, req *pb.UserRequest) (*pb.UserResponse, error) {
+	//TODO
+	logRequestService("ShareDocument")
+	return nil, nil
+}
+
+func logRequestService (svc string) {
```
Comment (faraonc) [inline service/service.go]: Please see comment above - https://github.com/faraonc/hwsc-logging

File: logtag/logtag.go (hunk 1 above)
Comment (faraonc) [reply logtag/logtag.go]: We can add this as a `dep` package.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/faraonc/hwsc-user-svc/logtag"
+	"google.golang.org/grpc"
+	"log"
+	"net"
+	pb "github.com/faraonc/hwsc-api-blocks/int/hwsc-user-svc/proto"
+	svc "github.com/faraonc/hwsc-user-svc/service"
+)
+
+const (
+	connectType = "tcp"
```
Comment (faraonc) [inline main.go]: This is on me, but a configuration management will be needed in the future. We might as well start doing this practice. Please review this PR. https://github.com/faraonc/hwsc-document-svc/pull/5

File: service/service.go (hunk 2 above)
Comment (faraonc) [reply service/service.go]: Make sure you implement something similar to ``fmt.Printf()`` for formatting strings. In that way we can simplify our func parameters.
https://gobyexample.com/string-formatting

There is a concept called variadic functions in GoLang.
https://gobyexample.com/variadic-functions

This is worth looking into as well for building Strings. 
https://golang.org/pkg/fmt/#Sprintf

File: service/service.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	log.Println(logtag.Info, "Requesting", svc, "service")
```
Comment (faraonc) [inline service/service.go]: ``Printf`` would be an ideal for this case.

File: logtag/logtag.go (hunk 1 above)
Comment (faraonc) [reply logtag/logtag.go]: I can already predict logging will be big. We can hook it up with something like https://prometheus.io/ or https://grafana.com/ later.

File: conf/config.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+// set up some form of state on the initial startup of our program
+// i.e: creating c/n to DBs, loading config files, initializing variable, etc.
+// regardless of how many times package is imported, init is only called once
+func init() {
+	// config.Load(): Load config from a file source
+	// file extension determines config format
+	// in this case, load json config file
+	var err error
+	err = config.Load(file.NewSource(file.WithPath(confFilePath)))
+	if err != nil {
```
Comment (faraonc) [inline conf/config.go]: Just take note if you are having issues with unit test

File: conf/config.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	// and copy all "grpc-server" properties to GRPCHost (type struct Host)
+	// &GRPCHost = reference to GRPCHost (address), similar to pass-by-reference
+	err = config.Get("hosts", "grpc-server").Scan(&GRPCHost)
+	if err != nil {
+		log.Fatal("Failed to scan conf file", err.Error())
+	}
+
+	// scan "hosts" prop "mongodb-document" properties from config file
+	// and copy its props to UserDB struct
+	err = config.Get("hosts", "mongodb-document").Scan(&UserDB)
```
Comment (faraonc) [inline conf/config.go]: mongodb-user

**Pull request merged**